go/adbc/sqldriver/flightsql/flightsql_test.go

```diff
@@ -0,0 +1,141 @@
+	"github.com/apache/arrow/go/v12/arrow/flight/flightsql"
+	"github.com/apache/arrow/go/v12/arrow/flight/flightsql/example"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/grpc"
+	"os"
+	"testing"
+)
+
+func Example() {
+	// Be sure to import the driver first:
+	// import _ "github.com/apache/arrow-adbc/go/adbc/sqldriver/flightsql"
+
+	db, err := sql.Open("flightsql", "uri=grpc://localhost:12345")
+	if err != nil {
+		panic(err)
+	}
+
+	if err = db.Ping(); err != nil {
+		panic(err)
+	}
+}
+
+type SQLDriverFlightSQLSuite struct {
+	suite.Suite
+
+	srv      *example.SQLiteFlightSQLServer
+	s        flight.Server
+	opts     []grpc.ServerOption
+	sqliteDB *sql.DB
+
+	done chan bool
+	mem  *memory.CheckedAllocator
+}
+
+func (suite *SQLDriverFlightSQLSuite) SetupTest() {
+	var err error
+
+	suite.sqliteDB, err = example.CreateDB()
+	require.NoError(suite.T(), err)
+
+	suite.mem = memory.NewCheckedAllocator(memory.DefaultAllocator)
+	suite.s = flight.NewServerWithMiddleware(nil, suite.opts...)
+	require.NoError(suite.T(), err)
+	suite.srv, err = example.NewSQLiteFlightSQLServer(suite.sqliteDB)
+	require.NoError(suite.T(), err)
+	suite.srv.Alloc = suite.mem
+
+	suite.s.RegisterFlightService(flightsql.NewFlightServer(suite.srv))
+	require.NoError(suite.T(), suite.s.Init("localhost:0"))
+	suite.s.SetShutdownOnSignals(os.Interrupt, os.Kill)
+	suite.done = make(chan bool)
+	go func() {
+		defer close(suite.done)
+		_ = suite.s.Serve()
+	}()
+}
+
+func (suite *SQLDriverFlightSQLSuite) TearDownTest() {
+	if suite.done == nil {
+		return
+	}
+
+	suite.s.Shutdown()
+	<-suite.done
+	suite.srv = nil
+	suite.mem.AssertSize(suite.T(), 0)
+	_ = suite.sqliteDB.Close()
+	suite.done = nil
+}
+
+func (suite *SQLDriverFlightSQLSuite) dsn() string {
+	return "uri=grpc+tcp://" + suite.s.Addr().String()
+}
+
+type srvQuery string
+
+func (s srvQuery) GetQuery() string { return string(s) }
+
+func (s srvQuery) GetTransactionId() []byte { return nil }
+
+func (suite *SQLDriverFlightSQLSuite) TestQuery() {
+	db, err := sql.Open("flightsql", suite.dsn())
+	require.NoError(suite.T(), err)
+	defer db.Close()
+
+	_, err = suite.srv.DoPutCommandStatementUpdate(
+		context.Background(), srvQuery("CREATE TABLE t (k, v)"))
+	require.NoError(suite.T(), err)
+	_, err = suite.srv.DoPutCommandStatementUpdate(
+		context.Background(), srvQuery("INSERT INTO t (k, v) VALUES ('one', 'alpha'), ('two', 'bravo')"))
+	require.NoError(suite.T(), err)
+
+	// TODO db.Exec() fails:
+	// Received unexpected error:
+	// Invalid State: SqlState: , msg: [Flight SQL Statement] must call Prepare before GetParameterSchema
+	//
+	//_, err = db.Exec("CREATE TABLE t (k, v)")
+	//require.NoError(suite.T(), err)
+	//defer db.Exec("DROP TABLE IF EXISTS t")
+	//_, err = db.Exec("INSERT INTO t (k, v) VALUES ('one', 'alpha'), ('two', 'bravo')")
+	//require.NoError(suite.T(), err)
```

Comment:
@zeroshade got any clue here?

Reply:
figured out the issue and pushed a fix, I also cleaned up the test a little (using `suite.` instead of having to pass `suite.T()` constantly).

Reply:
Cool, TIL `suite.Require()`.

I pushed another commit to fix the lint error and convert more uses of `suite.T()`.